ledger/cow.go

```diff
@@ -180,7 +184,7 @@ func (cb *roundCowState) checkDup(firstValid, lastValid basics.Round, txid trans
 }
 
 func (cb *roundCowState) txnCounter() uint64 {
-	return cb.lookupParent.txnCounter() + uint64(len(cb.mods.Txids))
+	return cb.lookupParent.txnCounter() + cb.txnCount
```

Comment:
```suggestion
	return cb.lookupParent.txnCounter() + uint64(len(cb.mods.Txids)) + uint64(len(cb.mods.innerTxns))
```
`cb.mods.innerTxns` can be a map of array of txn or txnids that is populated in a new method `addInnerTx` similar to `addTx`. The idea is to have actual data and not just a counter with non-obvious places to increment.

Reply:
Then we'd be maintaining a list, just so we can count it later.  It would have the same potential for error.  Wherever we are currently incrementing the count, we'd need to call `addInnerTx`.  But we don't need the list.

Reply:
the benefit is to make all calculations for nested cows work for free - one day teal can create app call txn that would spin-off another child cow for `StatefulEval` in `Perform`. Then propagating `innerTxns` in `commitParent` (or a counter for inner txns into parent's inner txns counter) would let `txnCounter()` func just work. No need to analyze inner txns from apply data in `eval.transaction` and so on.

Reply:
I think we're on the same page.  I did modify commitToParent(), so it does:
`cb.commitParent.txnCount += cb.txnCount`
That allows children to work properly, I think.

Reply:
I get your point now.  You want it be true, the whole time, that cb.txnCounter() is correct.  I was only incrementing the count after all inner transactions for a given top-level txn were done.  That's wrong, because I won't know the count after one is done, but I'm working on the next.  I'm changing it to increment the count after each inner transaction.